src/metpy/calc/boundarylayer.py

```diff
@@ -0,0 +1,305 @@
+#!/usr/bin/python
+# -*-coding:utf-8 -*-
+"""
+Contains a collection of boundary layer height estimations.
+
+
+References:
+-----------
+
+[Col14]: Collaud Coen, M., Praz, C., Haefele, A., Ruffieux, D., Kaufmann, P., and Calpini, B. (2014): 
+    Determination and climatology of the planetary boundary layer height above the Swiss plateau by in situ and remote sensing measurements as well as by the COSMO-2 model
+    Atmos. Chem. Phys., 14, 13205–13221.
+
+[HL06]: Hennemuth, B., & Lammert, A. (2006):
+    Determination of the atmospheric boundary layer height from radiosonde and lidar backscatter.
+    Boundary-Layer Meteorology, 120(1), 181-200.
+
+[Guo16]: Guo, J., Miao, Y., Zhang, Y., Liu, H., Li, Z., Zhang, W., ... & Zhai, P. (2016):
+    The climatology of planetary boundary layer height in China derived from radiosonde and reanalysis data.
+    Atmos. Chem. Phys, 16(20), 13309-13319.
+
+[Sei00]: Seidel, D. J., Ao, C. O., & Li, K. (2010):
+    Estimating climatological planetary boundary layer heights from radiosonde observations: Comparison of methods and uncertainty analysis.
+    Journal of Geophysical Research: Atmospheres, 115(D16).
+    
+[VH96]: Vogelezang, D. H. P., & Holtslag, A. A. M. (1996):
+    Evaluation and model impacts of alternative boundary-layer height formulations.
+    Boundary-Layer Meteorology, 81(3-4), 245-269.
+"""
+
+import numpy as np
+from copy import deepcopy
+
+import metpy.calc as mpcalc
+import metpy.constants as mpconsts
+from metpy.units import units
+
+
+def smooth(val, span):
+    """Function that calculates the moving average with a given span.
+    The span is given in number of points on which the average is made.
+
+    Parameters
+    ----------
+    val: array-like
+        Array of values
+    span: int
+        Span of the moving average. The higher the smoother
+
+    Returns
+    -------
+    smoothed_val: array-like
+        Array of smoothed values
+    """
+    N = len(val)
+    smoothed_val = deepcopy(val)
+    for i in range(N):
+        smoothed_val[i] = np.nanmean(val[i - min(span, i) : i + min(span, N - i)])
```

Comment:
```suggestion
        smoothed_val[i] = np.nanmean(val[max(i - span, 0) : min(i + span, N)])
```
Would this be more clear as to the intent?